submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestCleaner.kt

```diff
@@ -39,18 +40,28 @@ class SubmissionRequestCleaner(
         fun deleteFile(index: Int, file: ExtFile) {
             logger.info { "${current.accNo} ${current.owner} Deleting file $index, path='${file.filePath}'" }
             storageService.deleteSubmissionFile(current, file)
+            storageService.deleteFtpFile(current, file)
         }
 
+        fun shouldDelete(newFiles: Map<String, FileEntry>, existing: ExtFile): Boolean =
+            when (val newFile = newFiles[existing.filePath]) {
+                null -> false
+                else -> newFile.md5 != existing.md5 && new.storageMode == existing.storageMode
+            }
+
         val newFiles = newFilesMap(new)
         logger.info { "${current.accNo} ${current.owner} Started cleaning common submission files" }
         serializationService.fileSequence(current)
-            .filter { newFiles.containsKey(it.filePath) && newFiles[it.filePath] != it.md5 }
+            .filter { shouldDelete(newFiles, it) }
```

Comment:
previous mechanism delete file regardless if the storage type is the same or not resulting in deletion of importants files as page tab ones.